src/modules/deliveryoptimization/src/lib/DeliveryOptimization.c

```diff
@@ -0,0 +1,417 @@
+{
+    MMI_HANDLE handle = (MMI_HANDLE)g_deliveryOptimizationModuleName;
+    g_maxPayloadSizeBytes = maxPayloadSizeBytes;
+    ++g_referenceCount;
+    OsConfigLogInfo(DeliveryOptimizationGetLog(), "MmiOpen(%s, %d) returning %p", clientName, maxPayloadSizeBytes, handle);
+    return handle;
+}
+
+static bool IsValidSession(MMI_HANDLE clientSession)
+{
+    return ((NULL == clientSession) || (0 != strcmp(g_deliveryOptimizationModuleName, (char*)clientSession)) || (g_referenceCount <= 0)) ? false : true;
+}
+
+void DeliveryOptimizationMmiClose(MMI_HANDLE clientSession)
+{
+    if (IsValidSession(clientSession))
+    {
+        --g_referenceCount;
+        OsConfigLogInfo(DeliveryOptimizationGetLog(), "MmiClose(%p)", clientSession);
+    }
+    else 
+    {
+        OsConfigLogError(DeliveryOptimizationGetLog(), "MmiClose() called outside of a valid session");
+    }
+}
+
+int DeliveryOptimizationMmiGetInfo(const char* clientName, MMI_JSON_STRING* payload, int* payloadSizeBytes)
+{
+    int status = EINVAL;
+
+    if ((NULL == payload) || (NULL == payloadSizeBytes))
+    {
+        return status;
+    }
+    
+    *payloadSizeBytes = (int)strlen(g_deliveryoptimizationModuleInfo);
+    *payload = (MMI_JSON_STRING)malloc(*payloadSizeBytes);
+    if (*payload)
+    {
+        memcpy(*payload, g_deliveryoptimizationModuleInfo, *payloadSizeBytes);
+        status = MMI_OK;
+    }
+    else
+    {
+        OsConfigLogError(DeliveryOptimizationGetLog(), "MmiGetInfo: failed to allocate %d bytes", *payloadSizeBytes);
+        *payloadSizeBytes = 0;
+        status = ENOMEM;
+    }
+    
+    if (IsFullLoggingEnabled())
+    {
+        OsConfigLogInfo(DeliveryOptimizationGetLog(), "MmiGetInfo(%s, %.*s, %d) returning %d", clientName, *payloadSizeBytes, *payload, *payloadSizeBytes, status);
+    }
+
+    return status;
+}
+
+int DeliveryOptimizationMmiGet(MMI_HANDLE clientSession, const char* componentName, const char* objectName, MMI_JSON_STRING* payload, int* payloadSizeBytes)
+{
+    int status = MMI_OK;
+
+    if ((NULL == componentName) || (NULL == objectName) || (NULL == payload) || (NULL == payloadSizeBytes))
+    {
+        OsConfigLogError(DeliveryOptimizationGetLog(), "MmiGet(%s, %s, %p, %p) called with invalid arguments", componentName, objectName, payload, payloadSizeBytes);
+        status = EINVAL;
+        return status;
+    }
+
+    *payload = NULL;
+    *payloadSizeBytes = 0;
+
+    if (!IsValidSession(clientSession))
+    {
+        OsConfigLogError(DeliveryOptimizationGetLog(), "MmiGet(%s, %s) called outside of a valid session", componentName, objectName);
+        status = EINVAL;
+    }
+        
+    if ((MMI_OK == status) && (strcmp(componentName, g_deliveryOptimizationComponentName)))
+    {
+        OsConfigLogError(DeliveryOptimizationGetLog(), "MmiGet called for an unsupported component name (%s)", componentName);
+        status = EINVAL;
+    }
+
+    const char* jsonPropertyName = NULL;
+    int jsonPropertyType = JSONNull;

```

Comment:
All locals need to be declared at top of function for C. For consistency with rest of project please follow this old C rule even if it's not enforced by compiler anymore.

There are many instances of this.

Please do not declare locals in C code using C++ style.

Reply:
Thanks, fixed!